mindsdb_server/namespaces/datasource.py

```diff
@@ -208,97 +308,147 @@ class Analyze(Resource):
-            where = []
-            for key, value in request.args.items():
-                if key == 'page[size]':
-                    params['page[size]'] = int(value)
-                if key == 'page[offset]':
-                    params['page[offset]'] = int(value)
-                elif key.startswith('filter'):
-                    result = re.search(r'filter\[(.*)\]', key)
-                    field = result.groups(1)[0]
-                    where.append({'field': field, 'value': value})
-            if params['page[size]'] is not None:
-                limit = f"limit {params['page[size]']}"
-            if params['page[size]'] is not None and params['page[offset]'] is not None:
-                offset = f"offset {params['page[offset]']}"
-
-            con = sqlite3.connect(db_path)
-            cur = con.cursor()
-            cur.execute('pragma table_info(data);')
-            column_name_index = [x[0] for x in cur.description].index('name')
-            columns = cur.fetchall()
-            column_names = [x[column_name_index] for x in columns]
-            where = [x for x in where if x['field'] in column_names]
-            marks = {}
-            if len(where) > 0:
-                for i in range(len(where)):
-                    field = where[i]["field"].replace('"', '""')
-                    if ' ' in field:
-                        field = f'"{field}"'
-                    marks['var' + str(i)] = '%' + where[i]['value'] + '%'
-                    where[i] = f'{field} like :var{i}'
-                where = 'where ' + ' and '.join(where)
-            else:
-                where = ''
-            count_query = ' '.join(['select count(1) from data', where])
-            query = ' '.join(['select * from data', where, order, limit, offset])
-
-            cur.execute(count_query, marks)
-            rowcount = cur.fetchone()[0]
-            cur.execute(query, marks)
-            data = cur.fetchall()
-            data = [dict(zip(column_names, x)) for x in data]
-            cur.close()
-            con.close()
-
-            response = {
-                'rowcount': rowcount,
-                'data': data
-            }
-            return response, 200
-        abort(404, "")
+        ds = get_datasource(name)
+        if ds is None:
+            print('No valid datasource given')
+            abort(400, 'No valid datasource given')
+
+        ds_dir = os.path.join(mindsdb.CONFIG.MINDSDB_DATASOURCES_PATH, ds['name'], 'datasource')
+        db_path = os.path.join(ds_dir, 'sqlite.db')
+        if os.path.isfile(db_path) is False:
+            path = ds['source']
+            if ds['source_type'] == 'file':
+                if not os.path.exists(path):
+                    abort(404, "")
+            ds = FileDS(path)
+            
+            df_with_types = cast_df_columns_types(ds.df)
+            create_sqlite_db(os.path.join(ds_dir, 'sqlite.db'), df_with_types)
+
+        limit = ''
+        offset = ''
+        order = ''
```

Comment:
Bit tidier to delcare these 3 later in a one line if else.

Example for `limit`:

`limit = '' if params['page[size]'] is None else  f"limit {params['page[size]']}"`

Reply:
done